src/features/draw-rolling-circle/Drawing.js

```diff
@@ -12,7 +12,7 @@ const initialState = {
     theta: Math.PI / 2,
     direction: 1,
     speed: Physics.VELOCITY,
-    centerX: radius,
+    centerX: radius + 1,
```

Comment:
I can imagine a situation where the circle will still end up just too far past the edge that the direction will flip each time again.  Instead of changing the starting position, the logic that switches direction should not just flip sign, but rather set direction = 1 or direction = -1.

Reply:
I assumed that's what was happening, causing the bug. You can see the ball wiggling left/right, which smells like a direction change. I haven't seen the collision detection fail outside of the initial position. If we think it's possible we could pad things.

Reply:
No, I mean hard code the direction change.  Then you don't need to mess with padding or hopes or dreams!
```javascript
if (rollingCircle.centerX >= WIDTH - radius) {
  rollingCircle.direction = -1
}
if (rollingCircle.centerX <= radius) {
  rollingCircle.direction = 1
}
```